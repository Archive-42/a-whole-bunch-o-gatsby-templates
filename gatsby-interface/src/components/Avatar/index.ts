import { AvatarSize as AvatarSizeDefinition } from "./types"
import { AvatarProps as AvatarPropsDefinition } from "./Avatar"
import {
  AvatarsGroupProps as AvatarsGroupPropsDefinition,
  AvatarDescriptor as AvatarDescriptorDefinition,
} from "./AvatarsGroup"

export type AvatarSize = AvatarSizeDefinition
export type AvatarProps = AvatarPropsDefinition
export type AvatarsGroupProps = AvatarsGroupPropsDefinition
export type AvatarDescriptor = AvatarDescriptorDefinition

export { default as Avatar } from "./Avatar"
export { default as AvatarsGroup } from "./AvatarsGroup"
