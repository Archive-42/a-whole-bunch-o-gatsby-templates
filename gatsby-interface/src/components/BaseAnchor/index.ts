import { BaseAnchorProps as BaseAnchorPropsDefinition } from "./BaseAnchor"

export type BaseAnchorProps = BaseAnchorPropsDefinition

export { BaseAnchor } from "./BaseAnchor"
