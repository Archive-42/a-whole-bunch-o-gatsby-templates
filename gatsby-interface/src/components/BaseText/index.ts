import { BaseTextProps as BaseTextPropsDefinition } from "./BaseText"

export type BaseTextProps = BaseTextPropsDefinition

export { default as BaseText } from "./BaseText"
