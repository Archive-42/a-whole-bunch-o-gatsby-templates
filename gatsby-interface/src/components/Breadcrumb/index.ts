import { BreadcrumbProps as _BreadcrumbProps } from "./Breadcrumb"

export type BreadcrumbProps = _BreadcrumbProps

export { default as Breadcrumb } from "./Breadcrumb"
