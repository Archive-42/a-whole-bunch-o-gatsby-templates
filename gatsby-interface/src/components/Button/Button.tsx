/** @jsx jsx */
import { jsx, InterpolationWithTheme } from "@emotion/core"
import React from "react"
import { MdRefresh } from "react-icons/md"

import { BaseButton, BaseButtonProps } from "../BaseButton"
import {
  getButtonCss,
  ButtonVariant,
  ButtonTone,
  ButtonSize,
} from "../../theme/styles/button"
import { Theme } from "../../theme"

export type ButtonStyleProps = {
  size?: ButtonSize
  tone?: ButtonTone
  variant?: ButtonVariant
  leftIcon?: React.ReactNode
  rightIcon?: React.ReactNode
}

export type ButtonProps = BaseButtonProps & ButtonStyleProps

export function getButtonStyles({
  children,
  size = `L`,
  tone = `BRAND`,
  variant = `PRIMARY`,
  loading,
  leftIcon,
  rightIcon,
}: {
  children: React.ReactNode
  loading?: boolean
} & ButtonStyleProps): {
  css: InterpolationWithTheme<Theme>
  children: React.ReactNode
} {
  return {
    css: getButtonCss({
      size,
      variant,
      tone,
      loading,
      leftIcon,
      rightIcon,
    }),
    children:
      leftIcon || rightIcon ? (
        <React.Fragment>
          {leftIcon}
          {children}
          {rightIcon}
        </React.Fragment>
      ) : (
        children
      ),
  }
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (props, ref) => {
    const {
      children,
      loading,
      LoadingIcon = MdRefresh,
      size,
      tone,
      variant,
      leftIcon,
      rightIcon,
      ...rest
    } = props

    return (
      <BaseButton
        {...getButtonStyles({
          children,
          loading,
          size,
          tone,
          variant,
          leftIcon,
          rightIcon,
        })}
        loading={loading}
        LoadingIcon={LoadingIcon}
        {...rest}
        ref={ref}
      />
    )
  }
)
