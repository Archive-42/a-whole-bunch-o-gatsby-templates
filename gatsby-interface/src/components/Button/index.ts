export * from "./Buttons"
export * from "./Button"
