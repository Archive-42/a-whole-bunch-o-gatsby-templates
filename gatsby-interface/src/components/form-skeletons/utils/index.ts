export function getFinalAriaDescribedBy(
  controlDescribedBy?: string,
  ariaDescribedBy?: string
) {
  return (
    [controlDescribedBy, ariaDescribedBy]
      .filter(describedBy => describedBy)
      .join(` `) || undefined
  )
}

export function getHintId(fieldId: string) {
  return `${fieldId}__hint`
}

export function getErrorId(fieldId: string) {
  return `${fieldId}__error`
}
