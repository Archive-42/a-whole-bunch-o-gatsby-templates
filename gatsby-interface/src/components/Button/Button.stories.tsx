/** @jsx jsx */
import { jsx } from "@emotion/core"

import { storiesOf } from "@storybook/react"
import { action } from "@storybook/addon-actions"

import { Button } from "./Button"
import {
  PrimaryButton,
  SecondaryButton,
  CancelButton,
  SuccessButton,
  TextButton,
  PrimaryDeleteButton,
  SecondaryDeleteButton,
} from "./"

import README_MAIN from "./README_MAIN.md"
import README_ICONS from "./README_ICONS.md"
import customStyling from "./README_customStyling.md"
import README_MANUAL_STYLING from "./README_MANUAL_STYLING.md"
import README_LEGACY from "./README_LEGACY.md"
import { getButtonCss } from "../../theme/styles/button"
import { StoryUtils } from "../../utils/storybook"
import {
  showcaseVariants,
  showcaseSizes,
  showcaseTones,
  showcaseCustomStyles,
  showcaseIcons,
} from "./utils/storybook-styles"
import { MdCloud } from "react-icons/md"

storiesOf(`buttons/Button`, module)
  .addParameters({
    options: {
      showPanel: true,
    },
    readme: {
      sidebar: README_MAIN,
      includePropTables: [Button],
    },
  })
  .add(...showcaseVariants(Button, {}))
  .add(...showcaseSizes(Button, {}))
  .add(...showcaseTones(Button, {}))
  .add(...showcaseCustomStyles(Button, {}, customStyling))
  .add(...showcaseIcons(Button, {}, README_ICONS))
  .add(`in 'loading' state`, () => (
    <StoryUtils.Container>
      <StoryUtils.Stack>
        <Button onClick={action(`Button was clicked`)} loading={true}>
          Button in loading state
        </Button>
        <Button
          onClick={action(`Button was clicked`)}
          loading={true}
          loadingLabel={`Custom loading label`}
        >
          Button in loading state
        </Button>
      </StoryUtils.Stack>
    </StoryUtils.Container>
  ))
  .add(
    `custom button component`,
    () => {
      function CustomButton(props: JSX.IntrinsicElements["button"]) {
        return <button {...props} onClick={action("click custom component")} />
      }

      return (
        <StoryUtils.Container>
          <StoryUtils.Stack>
            <Button ButtonComponent={CustomButton} rightIcon={<MdCloud />}>
              Custom component
            </Button>
          </StoryUtils.Stack>
        </StoryUtils.Container>
      )
    },
    {
      readme: {
        sidebar: README_MANUAL_STYLING,
      },
    }
  )
  .add(
    `manually applied styles`,
    () => (
      <StoryUtils.Container>
        <StoryUtils.Stack>
          <button css={getButtonCss({ size: `L`, tone: `BRAND` })}>
            I'm a &lt;button&gt; but I look like the &lt;Button&gt;
          </button>
        </StoryUtils.Stack>
      </StoryUtils.Container>
    ),
    {
      readme: {
        sidebar: README_MANUAL_STYLING,
      },
    }
  )
  .add(
    `legacy Buttons`,
    () => (
      <StoryUtils.Container>
        <StoryUtils.Stack>
          <PrimaryButton>PrimaryButton</PrimaryButton>
          <SecondaryButton>SecondaryButton</SecondaryButton>
          <CancelButton>CancelButton</CancelButton>
          <SuccessButton>SuccessButton</SuccessButton>
          <TextButton>TextButton</TextButton>
          <PrimaryDeleteButton>PrimaryDeleteButton</PrimaryDeleteButton>
          <SecondaryDeleteButton>SecondaryDeleteButton</SecondaryDeleteButton>
        </StoryUtils.Stack>
      </StoryUtils.Container>
    ),
    {
      readme: {
        sidebar: README_LEGACY,
      },
    }
  )
