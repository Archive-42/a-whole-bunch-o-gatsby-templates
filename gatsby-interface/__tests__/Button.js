import React from "react"
import { renderWithTheme } from "../src/utils/testing"

import { PrimaryButton } from "../src/components/Button"

const render = renderWithTheme

describe(`<PrimaryButton>`, () => {
  test(`renders unchanged`, async () => {
    const { container } = render(<PrimaryButton>Click me!</PrimaryButton>)

    expect(container).toMatchSnapshot()
  })
})
