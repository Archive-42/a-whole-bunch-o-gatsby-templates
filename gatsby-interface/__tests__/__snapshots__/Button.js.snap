// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton> renders unchanged 1`] = `
<div>
  <button
    class="css-9qeu11"
    type="button"
  >
    Click me!
  </button>
</div>
`;
